components/logins/src/error.rs

```diff
@@ -21,9 +21,21 @@ pub enum LoginsApiError {
     #[error("No record with guid exists (when one was required): {reason:?}")]
     NoSuchRecord { reason: String },
 
+    #[error("Encryption key is missing.")]
+    MissingKey,
+
+    #[error("Encryption key is not valid.")]
+    InvalidKey,
+
     #[error("Encryption key is in the correct format, but is not the correct key.")]
```

Comment:
It's not immediately clear how "incorrect key" works with the 2 new "Failed" error cases? I wonder if that can just be replaced with the new error values?

Reply:
I think its used only for the `check_canary` function. To me it sounds a little strange that it throws if the key does not match - isn't it intended to just return false in that case?

Reply:
> I think its used only for the `check_canary` function. To me it sounds a little strange that it throws if the key does not match - isn't it intended to just return false in that case?

Yeah, I'd expect a valid but incorrect key to just return false there instead of an err (or *only* return the Err without any bool)

Reply:
ok cool, will make this change, since we're breaking anyways :)